### PR TITLE
More roundabout phrase tests

### PR DESCRIPTION
**Pull request by dgearhart**
# Issue

- Add `Signs::GetGuideString` unit tests
- Fix `Signs::GetGuideString` to properly round or truncate max_count for each sign type
- Add roundabout on unnamed cycleway test
- Pass in the four expected instructions to the gurka `expect_instructions_at_maneuver_index` method 

## Tasklist

 - [x] Add tests
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

### CODE REVIEW

File: src/odin/signs.cc [hunk 1]
```diff
@@ -138,12 +143,18 @@ const std::string Signs::GetGuideString(uint32_t max_count,
...
                                         const VerbalTextFormatter* verbal_formatter) const {
   std::string guide_string;
+  // If both branch and toward exist
+  // and either unlimited max count or max count is greater than 1
+  // then process guide sign info splitting between branch and toward signs
   if (HasGuideBranch() && HasGuideToward() && ((max_count == 0) || (max_count > 1))) {
+    // Round using floating point division
     std::string guide_branch =
-        GetGuideBranchString(static_cast<uint32_t>(std::round(max_count / 2)),
+        GetGuideBranchString(static_cast<uint32_t>(
```
Comment (dgearhart) [inline src/odin/signs.cc:152]: Thank you @mandeepsandhu for flagging the integer division

File: test/signs.cc [hunk 2]
```diff
@@ -243,6 +260,71 @@ TEST(Signs, TestGetGuideTowardString_roundabout_toward_A1) {
   TryGetGuideTowardString(signs, 1, true, "A 1");
 }
 
+TEST(Signs, TestGetGuideString_BranchOnly) {
+  // Create guide sign
+  // Specify input in descending consecutive count order
+  Signs signs =
+      GetGuideSigns({std::make_tuple("US 322 West", 1, 1), std::make_tuple("US 22 West", 1, 0),
```
Comment (mandeepsandhu) [inline test/signs.cc:267]: Can we have a test case where the consecutive count is > 1, say 2 of them have `2` and the result shows both items getting selected when `limit_by_consecutive_count` is set (for different values of. `max_count`)?

Comment (dgearhart) [reply test/signs.cc:267]: @mandeepsandhu pushed up more tests
Added more TestGetGuideString tests with no, 1, and multiple consecutive counts

File: test/signs.cc [hunk 3]
```diff
@@ -243,6 +260,123 @@ TEST(Signs, TestGetGuideTowardString_roundabout_toward_A1) {
...
+      "US 322 West/US 22 West/Freedom Highway/Valhalla Highway/A 1/Remscheid/Wermelskirchen/Hückeswagen");
+  TryGetGuideString(signs, 4, false, "US 322 West/US 22 West/A 1/Remscheid");
+  TryGetGuideString(signs, 3, false, "US 322 West/US 22 West/A 1");
+  TryGetGuideString(signs, 2, false, "US 322 West/A 1");
+  TryGetGuideString(signs, 1, false, "US 322 West");
+
+  TryGetGuideString(
+      signs, 0, true,
+      "US 322 West/US 22 West/Freedom Highway/Valhalla Highway/A 1/Remscheid/Wermelskirchen/Hückeswagen");
+  TryGetGuideString(signs, 4, true, "US 322 West/US 22 West/A 1/Remscheid");
```
Comment (mandeepsandhu) [inline test/signs.cc:324]: Why is `Remscheid` being returned here? I thought if `limit_by_consecutive_count` is set, only entries with the same consecutive count are returned? In this case `A 1` has a consecutive count of 1, while `Remscheid`'s is 0, which should make us drop `Remscheid`, no?

Comment (mandeepsandhu) [reply test/signs.cc:324]: > Why is `Remscheid` being returned here? I thought if `limit_by_consecutive_count` is set, only entries with the same consecutive count are returned? In this case `A 1` has a consecutive count of 1, while `Remscheid`'s is 0, which should make us drop `Remscheid`, no?

Sorry, ignore this question. Thats not consecutive count I was looking at, but the is route attribute.

**Pull request merged**